server/methods/accounts/accounts.js

```diff
@@ -449,57 +461,49 @@ export function addressBookUpdate(address, accountUserId, type) {
-    // Cart should exist to this moment, so we doesn't need to to verify its
-    // existence.
+
+    // If isShippingDefault address has changed
     if (oldAddress.isShippingDefault !== address.isShippingDefault) {
-      // if isShippingDefault was changed and now it is `true`
+      // Update the cart to use new default shipping address
       if (address.isShippingDefault) {
-        // we need to add this address to cart
         Meteor.call("cart/setShipmentAddress", cart._id, address);
-        // then, if another address was `ShippingDefault`, we need to unset it
-        Accounts.update({
-          userId,
-          "profile.addressBook.isShippingDefault": true
-        }, {
-          $set: {
-            "profile.addressBook.$.isShippingDefault": false
-          }
-        });
-        Hooks.Events.run("afterAccountsUpdate", Meteor.userId(), account._id);
       } else {
-        // if new `isShippingDefault` state is false, then we need to remove
-        // this address from `cart.shipping`
+        // If the new address is not the shipping default, remove it from the cart
         Meteor.call("cart/unsetAddresses", address._id, userId, "shipping");
       }
     } else if (address.isShippingDefault && oldAddress.isShippingDefault) {
-      // If current Shipping Address was edited but not changed update it to cart too
+      // If shipping address was edited, but isShippingDefault status not changed, update the cart address
       Meteor.call("cart/setShipmentAddress", cart._id, address);
     }
 
-    // the same logic used for billing
+    // If isBillingDefault address has changed
     if (oldAddress.isBillingDefault !== address.isBillingDefault) {
+      // Update the cart to use new default billing address
       if (address.isBillingDefault) {
         Meteor.call("cart/setPaymentAddress", cart._id, address);
-        Accounts.update({
-          userId,
-          "profile.addressBook.isBillingDefault": true
-        }, {
-          $set: {
-            "profile.addressBook.$.isBillingDefault": false
-          }
-        });
-        Hooks.Events.run("afterAccountsUpdate", Meteor.userId(), account._id);
       } else {
+        // If the new address is not the shipping default, remove it from the cart
         Meteor.call("cart/unsetAddresses", address._id, userId, "billing");
       }
     } else if (address.isBillingDefault && oldAddress.isBillingDefault) {
-      // If current Billing Address was edited but not changed update it to cart too
+      // If shipping address was edited, but isShippingDefault status not changed, update the cart address
       Meteor.call("cart/setPaymentAddress", cart._id, address);
     }
   }
 
+  // Update all addresses to set new default address to true, and the rest to false
+  account.profile.addressBook.forEach((addr) => {
+    if (addr._id === address._id) {
+      Object.assign(addr, { [type]: true });
+    } else {
+      Object.assign(addr, { [type]: false });
+    }
+  });
```

Comment:
This changes the behavior a bit from before, in that you are no longer updating the whole address object where `addr._id === address._id`. Since you've already done `Object.assign(address, { [type]: true });` above, I think you can do this:

```js
account.profile.addressBook.forEach((addr) => {
  if (addr._id === address._id) {
    Object.assign(addr, address);
  } else if (typeof type === "string") {
    Object.assign(addr, { [type]: false });
  }
});
```

Note that I also added a check for whether `type` is a string to the `else`. That's because it seems like this is intended to work without passing a type also, if you just want to update the address.